pkg/common/cns-lib/volume/util.go

```diff
@@ -362,21 +362,28 @@ func ExtractFaultTypeFromErr(ctx context.Context, err error) string {
 func ExtractFaultTypeFromVolumeResponseResult(ctx context.Context,
 	resp *cnstypes.CnsVolumeOperationResult) string {
 	log := logger.GetLogger(ctx)
+	log.Debugf("Extracting fault type from response: %+v", resp)
 	var faultType string
 	fault := resp.Fault
 	if fault != nil {
 		// faultType has the format like "*type.XXX", XXX is the specific VimFault type.
 		// For example, when CnsVolumeOperationrResult failed with ResourceInUse, faultType will be "*type.ResourceInUse".
-		faultType = reflect.TypeOf(fault.Fault).String()
-		log.Infof("Extract vimfault type: +%v  vimFault: +%v Fault: %+v from resp: +%v",
-			faultType, fault.Fault, fault, resp)
-		slice := strings.Split(faultType, ".")
-		vimFaultType := vimFaultPrefix + slice[1]
-		return vimFaultType
+		if fault.Fault != nil {
+			faultType = reflect.TypeOf(fault.Fault).String()
+			log.Infof("Extract vimfault type: %+v  vimFault: %+v Fault: %+v from resp: %+v",
+				faultType, fault.Fault, fault, resp)
+			slice := strings.Split(faultType, ".")
+			vimFaultType := vimFaultPrefix + slice[1]
+			return vimFaultType
+		} else {
+			faultType = reflect.TypeOf(fault).String()
```

Comment:
Is this needed due to some recent changes in the CNS backend ?
Asking because, I wanted to confirm if this change will work on all versions of VC.

Reply:
No, this change was required because while deleting Volumes from the MOB(as seen in the screenshot attached to PR https://bugzilla.eng.vmware.com/show_bug.cgi?id=3064078), the controller pods went into CLBO state and it was erroring out on the the line - build/pkg/common/cns-lib/volume/util.go:370
```
2022-12-02T21:55:41.226267550Z sigs.k8s.io/vsphere-csi-driver/v2/pkg/common/cns-lib/volume.ExtractFaultTypeFromVolumeResponseResult({0x269de18?, 0xc00075bec0?}, 0xc00087c2a0)
2022-12-02T21:55:41.226739401Z  /build/pkg/common/cns-lib/volume/util.go:370 +0x95
```

because the type of Fault was not being populated by the caller. So, this check needs to be added and the FaultType needs to be extracted from the fault thrown. 
The Fault is thrown and is expected because we deleted volumes.